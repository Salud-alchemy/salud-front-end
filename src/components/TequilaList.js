import React, { Component } from 'react';
import request from 'superagent';
//import { Link } from 'react-router-dom';
import CocktailItem from './CocktailItem';




export default class TequilaList extends Component {
    state = {
        id: [],
        cocktail:[],
        favorites:[]
    }
    async componentDidMount() {

        const user = JSON.parse(localStorage.getItem('user'))
        const getTequilaList = () => request.get(`https://mighty-plateau-34350.herokuapp.com/tequila`)
            .set('Authorization', user.token);

        const getFavorites = () => request.get('https://mighty-plateau-34350.herokuapp.com/favorites')
            .set('Authorization', user.token); 

        
    const fave = await getFavorites()        
    const data = await getTequilaList()
    console.log(data.body)
    this.setState({
        cocktail: data.body,
        favorites: fave.body
    })
}

makeFavorite = async (drink) => {
    console.log('this works', drink)
    // when the user clicks the makeFavorite button, add this drink to the favorite list
    const user = JSON.parse(localStorage.getItem('user'))
    const fave = await request.post('https://mighty-plateau-34350.herokuapp.com/favorites', {
        name: drink.name,
        image: drink.image,
        api_id: drink.id 
       
    })
    .set('Authorization', user.token)
    console.log('fave', fave.body)
}
renderButton = (drink) => {
    // check the favorites list if we're on the search page
    const isOnFavoritesList = this.state.favorites.find(cocktails => drink.name === cocktails.name);
    if (!isOnFavoritesList) {
    
    return <button onClick={ () => this.makeFavorite(drink)}>Make favorite</button>
    }
    // otherwise, indicate that they ae already on the favorites list
    // this.makeFavorite(drink.id)
    return <span>🍸</span>
}

    render() {
        return (
            <div>
                <h2>Tequila Cocktails for fun and pleasure</h2>
                <ul className='tequila-list'>
                {
                        this.state.cocktail.map(cocktail => (
                            <>
                            <div key={cocktail.id} to={`/id/${cocktail.id}`} onClick={()=> this.props.history.push(`/id/${cocktail.id}`)}> 
                    
                            <CocktailItem cocktail={cocktail}/>
                            </div>
                            {this.renderButton(cocktail)}
                    </>
                        )
                        )   
                        }
                </ul>
            </div>
        )
    }
}