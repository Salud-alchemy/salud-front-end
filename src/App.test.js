
import React from 'react';
// import App from './App';
import renderer from 'react-test-renderer';
// import VodkaList from './components/VodkaList.js';
// import Popular from './components/Popular.js';
import TequilaList from './components/TequilaList';

// test('renders VodkaList correctly', () => {
//   const vodka = renderer
//     .create(<VodkaList />)
//     .toJSON();
//   expect(vodka).toMatchSnapshot();
// });

// test('renders popular list correctly', () => {
//   const popular = renderer
//     .create(<Popular />)
//     .toJSON();
//   expect(popular).toMatchSnapshot();
// });

test('renders tequila list correctly', () => {
  const tequila = renderer
    .create(<TequilaList />)
    .toJSON();
  expect(tequila).toMatchSnapshot();
});


