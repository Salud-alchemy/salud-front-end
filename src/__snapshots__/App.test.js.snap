// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Popular list correctly 1`] = `
<div>
  <h2>
    Vodka Cocktails for fun and pleasure
  </h2>
  <ul
    className="vodka-list"
  />
</div>
`;

exports[`renders VodkaList correctly 1`] = `
<div>
  <h2>
    Vodka Cocktails for fun and pleasure
  </h2>
  <ul
    className="vodka-list"
  />
</div>
`;

exports[`renders VodkaList correctly 2`] = `
<div>
  <h2>
    Vodka Cocktails for fun and pleasure
  </h2>
  <ul
    className="vodka-list"
  />
</div>
`;

exports[`renders VodkaList correctly 3`] = `
<div>
  <h2>
    Vodka Cocktails for fun and pleasure
  </h2>
  <ul
    className="vodka-list"
  />
</div>
`;

exports[`renders popular list correctly 1`] = `
<div>
  <h2>
    Popular cocktails to be the life of the party
  </h2>
  <ul
    className="popular-list"
  />
</div>
`;

exports[`renders popular list correctly 2`] = `
<div>
  <h2>
    Popular cocktails to be the life of the party
  </h2>
  <ul
    className="popular-list"
  />
</div>
`;

exports[`renders tequila list correctly 1`] = `
<div>
  <h2>
    Tequila Cocktails for fun and pleasure
  </h2>
  <ul
    className="tequila-list"
  />
</div>
`;

exports[`renders w list correctly 1`] = `
<div>
  <h2>
    Popular cocktails to be the life of the party
  </h2>
  <ul
    className="popular-list"
  />
</div>
`;

exports[`renders whis list correctly 1`] = `
<div>
  <h2>
    Popular cocktails to be the life of the party
  </h2>
  <ul
    className="popular-list"
  />
</div>
`;

exports[`renders whiskey list correctly 1`] = `
<div>
  <h2>
    Popular cocktails to be the life of the party
  </h2>
  <ul
    className="popular-list"
  />
</div>
`;
